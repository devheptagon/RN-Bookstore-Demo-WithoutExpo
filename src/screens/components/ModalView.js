import React from 'react'
import PropTypes from 'prop-types'
import Modal from 'react-native-modal'
import Theme from '../../constants/Theme'

const ModalView = ({ children, ...rest }) => (
    <Modal
        isVisible={true}
        backdropColor={Theme.Modal.BackgroundColor}
        backdropOpacity={0.3}
        {...rest}
    >
        {children}
    </Modal>
)

ModalView.propTypes = {
    children: PropTypes.any
}

export default ModalView