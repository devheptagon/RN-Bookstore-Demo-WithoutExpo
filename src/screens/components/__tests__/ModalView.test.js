import React from 'react';
import { Text } from 'react-native';
import ModalView from '../ModalView';
import renderer from 'react-test-renderer';

test('ModalView renders correctly', () => {
  const tree = renderer.create(<ModalView><Text>modal content</Text></ModalView>).toJSON();
  expect(tree).toMatchSnapshot();
});