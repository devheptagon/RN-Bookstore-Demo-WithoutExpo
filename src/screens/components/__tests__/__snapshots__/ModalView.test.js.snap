// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalView renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <Text>
      modal content
    </Text>
  </View>
</Modal>
`;
